ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -981,8 +994,44 @@ static VALUE rb_context_init_unsafe(VALUE self, VALUE isolate, VALUE snap) {
 
         Local<Context> context = Context::New(isolate_info->isolate);
 
+        Context::Scope context_scope { context };
+
         context_info->context = new Persistent<Context>();
         context_info->context->Reset(isolate_info->isolate, context);
+
+        Local<String> strClassName = String::NewFromUtf8Literal(isolate_info->isolate, "RubyError");
+
+        /* Create `RubyError` constructor + prototype (and steal `Error.captureStackTrace` while we're at it)
+         * equivalent JS code is roughly
+         * ```
+         * function RubyError() { throw new Error("This class may not be constructed from JS.") }
+         * RubyError.prototype = Object.create(Error.prototype);
+         * RubyError.prototype.name = "RubyError";
+         * RubyError.prototype.constructor = RubyError;
+         * ```
+         */
+        Local<Object> global = context->Global();
+        Local<Object> errorConstructor = global
+            ->Get(context, String::NewFromUtf8Literal(isolate_info->isolate, "Error"))
+            .ToLocalChecked().As<Object>();
```

Comment:
This is pretty early initialization code... plus, the existing mini_racer code was just as crash-prone.

I'd actually be surprised if any of this throws an error, though if you want that handled, that is going to be a bit more complicated, since we'd need a `v8::TryCatch` &c &c